File: iact-check-webapp/src/main/resources/webapp/src/app/shared/dtos/submission-dto.ts
import {RangeQuestionAnswerDTO} from "./range-question-answer-dto";
import {BookmarkedPossibleOutcomeDTO} from "./bookmarked-possible-outcome-dto";
import {ImageQuestionAnswerDTO} from "./image-question-answer-dto";
import {BookmarkedDocumentDTO} from "./bookmarked-document-dto";

export interface SubmissionDTO {
  id?: number;
  correlatingCheckId: number;
  firstName: string;
  lastName: string;
  street: string;
  zipCode: string;
  city: string;
  phoneNumber: string;
  email: string;
  rangeQuestionAnswers: RangeQuestionAnswerDTO[];
  imageQuestionAnswers: ImageQuestionAnswerDTO[];
  bookmarkedPossibleOutcomes: BookmarkedPossibleOutcomeDTO[];
  bookmarkedDocuments: BookmarkedDocumentDTO[];
}

File: iact-check-core/src/main/kotlin/ch/iact/iactcheck/dto/SubmissionDTO.kt
package ch.iact.iactcheck.dto

data class SubmissionDTO(
    val id: Long,
    val correlatingCheckId: Long,
    val firstName: String,
    val lastName: String,
    val street: String,
    val zipCode: String,
    val city: String,
    val phoneNumber: String,
    val email: String,
    val rangeQuestionAnswers: List<RangeQuestionAnswerDTO> = emptyList(),
    val bookmarkedPossibleOutcomes: List<BookmarkedPossibleOutcomeDTO> = emptyList()
)
